pkg/irsa/capa/capa.go

```diff
@@ -333,6 +336,45 @@ func (s *Service) Reconcile(ctx context.Context, outRequeueAfter *time.Duration)
 		return err
 	}
 
+	// We only need to create the OIDC provider if the workload cluster uses a different account than the management cluster.
+	// If they would use the same account, the OIDC provider would already be there.
+	if s.Scope.AccountID() != s.Scope.ManagementClusterAccountID() {
+		createMCOIDCProvider := func() error {
+			mcIdentityProviderURL := util.EnsureHTTPS(strings.Replace(key.CloudFrontAlias(s.Scope.BaseDomain()), s.Scope.ClusterName(), s.Scope.Installation(), 1))
+
+			s.Scope.Logger().Info("creating MC OIDC provider in WC AWS account", "identityProviderURLs", mcIdentityProviderURL)
+
+			i := &awsiam.CreateOpenIDConnectProviderInput{
+				Url:          aws.String(mcIdentityProviderURL),
+				ClientIDList: []*string{aws.String(key.STSUrl(s.Scope.Region()))},
+				Tags: []*awsiam.Tag{
+					{
+						Key:   aws.String("giantswarm.io/installation"),
+						Value: aws.String(s.Scope.Installation()),
+					},
+					{
+						Key:   aws.String("giantswarm.io/cluster"),
```

Comment:
This shouldn't be included I think. The MC can manage multiple WCs on the same account, but only one OIDC provider will be there for that MC. That also means that you should not error if the provider is already there.